### ISSUE TITLE
Add support for creating a database with option for choosing modes in datastore APIs

### ISSUE THREAD

**Issue by himanshu219** [googleapis/google-cloud-python#8211]
Currently there is no way to create a database in datastore via APIs and customers find it frustrating to go to the console create a database choose the mode in their project.

It would be good if we can automate this and provide APIs so that one can achieve this via gcloud sdk, python sdk etc.

**Comment by tseaver** [comment]
@himanshu219 I'm afraid I don't understand this request:  there is no "create a database" operation in the [Datastore API](https://cloud.google.com/datastore/docs/apis).  All entities for a given project are stored together, segregated only by `namespace` and their keys.  The Datastore panels in the Cloud Console do not expose such an operation, either.

Can you please point to docs for the feature you have in mind?

**Comment by himanshu219** [comment]
yes here are the docs https://cloud.google.com/datastore/docs/quickstart please refer to "Create a Database" section. Currently one has to go to the console and would like this step to be automated via APIs

**Comment by tseaver** [comment]
@himanshu219 OK, thanks.  I'm pretty sure that we *cannot* do this from the API:  the "create a database" option described there is how one enables using the Datastore API project:  like all other Google Cloud APIs, client libraries cannot do this step directly.

**Comment by himanshu219** [comment]
so does that mean in future also there are no plans to automate this? Also any particular reason why in general all other google cloud APIs also behave similarly? Thanks

**Comment by tseaver** [comment]
The client libraries are designed to wrap the methods exposed by the various APIs, but a prerequisite for using those methods is to enable the API, which is normally done through the console.  There may be a *different* API for managing the set of enabled APIs, but I'm not aware of it:  neither the [Billing Service API](https://cloud.google.com/billing/) nor the [Resource Manager API](https://cloud.google.com/resource-manager/) expose that facility.

If such an API exists, we would need to wrap it in a *different* library than `google-cloud-datastore`.
